src/tests/ftest/io/nvme_object.py

```diff
@@ -0,0 +1,139 @@
+from daos_api import DaosApiError
+from test_utils import TestPool, TestContainer
+from apricot import TestWithServers
+
+class NvmeObject(TestWithServers):
+    """Test class for NVMe storage by creating/Updating/Fetching
+       large number of objects.
+
+    Test Class Description:
+        Test the general functional operations of objects on nvme storage
+        i.e. Creation/Updating/Fetching for single pool and multiple pools.
+
+    :avocado: recursive
+    """
+
+    def setUp(self):
+        """Set Up nodes for each test case"""
+        super(NvmeObject, self).setUp()
+
+        # set common params
+        self.record_size = self.params.get("record_size", "/run/container/*")
+        self.pool_size = self.params.get("size", "/run/pool/createsize/*")
+
+    def tearDown(self):
+        """Tear Down each test case."""
+        try:
+            if self.container is not None:
+                self.container.destroy()
+            if self.pool is not None and self.pool.pool.attached:
+                self.pool.destroy(1)
+        finally:
+            # Stop the servers and agents
+            super(NvmeObject, self).tearDown()
+
+    @avocado.fail_on(DaosApiError)
+    def test_nvme_object_single_pool(self):
+        """Jira ID: DAOS-2087.
+
+        Test Description:
+            Test will create single pool on nvme using TestPool
+            Create large number of objects
+            Update/Fetch with different object ID in single pool
+
+        Use Cases:
+            Verify the objects are being created and the data is not
+            corrupted.
+        :avocado: tags=nvme,pr,hw,nvme_object_single_pool,small
+        """
+
+        # Test Params
+        self.pool = TestPool(self.context, self.log)
+        self.pool.get_params(self)
+        self.container = TestContainer(self.pool)
+        self.container.get_params(self)
+
+        # set pool size
+        self.pool.nvme_size.update(self.pool_size[0])
+        # Create a pool
+        self.pool.create()
+        self.pool.connect()
+
+        # create container
+        self.container.create()
+        print(self.record_size[:-1])
+        for record in self.record_size[:-1]:
+            self.container.record_qty.update(record)
+            print(self.container.record_qty)
+            # write multiple objects
+            self.container.write_objects()
+
+            # read written objects and verify
+            self.container.read_objects()
+
+    @avocado.fail_on(DaosApiError)
+    def test_nvme_object_multiple_pools(self):
+        """Jira ID: DAOS-2087.
+
+        Test Description:
+            Test will create multiple pools on nvme using TestPool
+            Create large number of objects for each pool
+            Update/Fetch with different object ID in multiple pools
+
+        Use Cases:
+            Verify the objects are being created and the data is not
+            corrupted.
+        :avocado: tags=nvme,full_regression,hw,nvme_object_multiple_pools,small
```

Comment:
If we are going to add more object it's not going to be small so lets see how long it takes to fill those pool and based on that either medium or large

Reply:
You still have small tag, should we change it to medium for second case?

Reply:
Yes I will change that